db_utils.py

```diff
@@ -95,15 +104,24 @@ def get_one_record(self, sql):
 
     def get_records(self, query):
         cursor = self.get_cursor()
-        cursor.execute(query)
+        try:
+            cursor.execute(query)
+        except Exception as e:
+            print(f"Exception thrown while processing sql: {query}\n{e}\n")
         record_list = list(cursor.fetchall())
         self.logger.debug(f"get records SQL: {query}")
         cursor.close()
         return record_list
 
-    def get_cursor(self):
+    def get_cursor(self, buffered=False):
         self.connect()
```

Comment:
This function is where we want a retry. right now your exception just masks the problem - adds logging but eats the exception and does nothign with it. 

Have it retry and then raise if the retry fails. If the retry succeeds, clear the "retry count"

Reply:
ok, I lied or you changed it... :-) self.connect covers this case. we can remove this logic and just return the cursor.